src/workloads/execution/FilterWithComplexLogicalExpression.yml

```diff
@@ -426,6 +434,535 @@ Actors:
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *HalfDocumentCount
+        BatchSize: 1000
+        Document:
+          c: 1
+          d: 1
+          a: { c: { d: 1 } }
+
+- Name: InsertDocumentsWithIgnoredPath
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [14]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *HalfDocumentCount
+        BatchSize: 1000
+        Document:
+          c: 1
+          d: 1
+          b: { c: { d: 1 } }
+
+- Name: QuiesceDocumentsWithTargetAndMissingPaths
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [15]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Both queries below will only target "a.c.d", skipping the documents which have "b.c.d".
+- Name: AggregationExpressionTargetPath
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [16]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$expr: {$eq: ["a.c.d", 0]}}
+
+- Name: MatchExpressionTargetPath
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [17]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {"a.c.d": 0}
+
+# Clean up the data from the previous phase.
+- Name: ClearCollectionWithTargetAndMissingPaths
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [18]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: QuiesceCollectionDropDocumentsWithTargetAndMissingPaths
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [19]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# Insert and query documents which have 3 levels of nested arrays across 3 path components.
+- Name: InsertNestedArrayData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [20]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {a: [
+          {^Repeat: {count: 3, fromGenerator: {b: [
+            {^Repeat: {count: 3, fromGenerator: {c: [1, 2, 3]}}}]}}}]}
```

Comment:
Can you add more elements to the a array that is a scalar or object doesn't contains field b?

Reply:
We can add them in both this and `InsertDeeplyNestedArrayData`, or have a separate test case. I think it's worth to test how we handle the missing field in array traversal.

Reply:
Adding a separate test case around this